chain-factory/src/factory.rs

```diff
@@ -1,7 +1,27 @@
 use chain_config::StakeMultiArg;
 
+multiversx_sc::derive_imports!();
 multiversx_sc::imports!();
 
+#[derive(TypeAbi, TopEncode, TopDecode, NestedEncode, NestedDecode, ManagedVecItem)]
+struct ContractMapArgs<M: ManagedTypeApi> {
+    id: ScArray,
+    address: ManagedAddress<M>,
+}
+
+#[derive(
+    TypeAbi, TopEncode, TopDecode, NestedEncode, NestedDecode, Clone, ManagedVecItem, PartialEq,
+)]
+pub enum ScArray {
+    None,
```

Comment:
Do you ever need a 0-value, like `None`? If not, I'd say simply remove it and also remove the check.

Reply:
The problem is to avoid the 0-value because it won't be saved in the mapper

Reply:
A 0 value is saved correctly in storage. Composed storage keys use nested encoding for members.

Reply:
removed it